core/main/src/firebolt/firebolt_gatekeeper.rs

```diff
@@ -142,8 +151,6 @@ impl FireboltGatekeeper {
                         trace!("check_with_roles for method ({}) succeded", request.method);
                     }
                 }
-            } else {
-                debug!("Role/Capability is cleared based on non-negotiable policy");
             }
         } else {
             // Couldnt find any capabilities for the method
```

Comment:
I think it's not correct to return error with DenyReason::Unsupported for the use case where all the required permission(s) are filtered out due to non-negotiable policy.  In this case, the permission check should return Ok(),
indicating that the required capabilities are permitted.